fix: CI build fails for fork PR

### Affected area

src/libs/Env.ts

### Describe the bug

The CI build fails on PRs from forks because the required environment variable NEXT_PUBLIC_CLERK_SIGN_IN_URL is empty/unavailable, causing zod validation in src/libs/Env.ts to abort the build.

## Run / Job

Actions run: https://github.com/Women-Devs-SG/bibsnbub/actions/runs/18787468059/job/53609221825
Job id: 53609221825
Run id: 18787468059

### Steps to reproduce

1. Open a pull request from a forked repository (external contributor).
2. GitHub Actions triggers the CI workflow (.github/workflows/CI.yml).
3. The job runs with environment: staging but secrets from environments are not exposed to fork PRs.
4. The build executes code that loads and validates env vars in src/libs/Env.ts and fails.

### Expected behavior

CI should either:
- skip secret-dependent build steps for forked PRs and run a lighter set of checks (lint, typecheck, tests that do not require secrets), or
- provide a clear, actionable error message explaining that secrets are not available for fork PRs.

## Actual
Build fails with zod validation error because NEXT_PUBLIC_CLERK_SIGN_IN_URL is empty/unset.

### Screenshots or recordings

## Relevant log excerpt
```bash
message: 'String must contain at least 1 character(s)',
path: [ 'NEXT_PUBLIC_CLERK_SIGN_IN_URL' ]
⨯ Failed to load next.config.ts
> Build error occurred
--- ERROR ---
Error: Invalid environment variables
    at Object.<anonymous> (src/libs/Env.ts:13:36)
```

## Relevant files
- Env validation: src/libs/Env.ts (ref: 42fde5fa02a43867bac51e69de72c575b94c6d7b) https://github.com/Women-Devs-SG/bibsnbub/blob/42fde5fa02a43867bac51e69de72c575b94c6d7b/src/libs/Env.ts
- CI workflow: .github/workflows/CI.yml (check how secrets/env are passed) https://github.com/Women-Devs-SG/bibsnbub/blob/42fde5fa02a43867bac51e69de72c575b94c6d7b/.github/workflows/CI.yml


### App/commit version

_No response_

### Environment

_No response_

### Priority

P0 - Critical (blocks release)

### Root cause

GitHub blocks repository and environment secrets from workflows that run untrusted code from forks. The workflow sets NEXT_PUBLIC_CLERK_SIGN_IN_URL from the environment secret (staging), but that secret is not available to forked PR runs, so the value is empty and validation fails.

### Suggested fixes / mitigation options (pick one)

1. Skip secret-requiring build on fork PRs (recommended)
- Run a lighter CI for fork PRs (lint, tests that do not need secrets).
- Run full secret-requiring builds only for:
-- pushes to branches in the repository
-- maintainer-triggered workflows (workflow_dispatch) or after a maintainer approves/merges.
- Example snippet to detect fork PRs and skip the build step:
 ```yaml
# Insert before running secret-dependent build steps
- name: Detect fork PR and skip secret-required build
  if: ${{ github.event_name == 'pull_request' && github.event.pull_request.head.repo.full_name != github.repository }}
  run: |
    echo "This PR comes from a fork. Environment secrets are not available; skipping secret-dependent build steps."
    exit 0
```

2. Fail early with a clearer message (less desirable but useful) 
```yaml
- name: Check presence of NEXT_PUBLIC_CLERK_SIGN_IN_URL
  run: |
    if [ -z "${NEXT_PUBLIC_CLERK_SIGN_IN_URL}" ]; then
      echo "##[error]NEXT_PUBLIC_CLERK_SIGN_IN_URL is missing or empty. If this is a forked PR, secrets are not exposed."
      exit 1
    fi
```

3. Use maintainer-run build for secrets
- After review, maintainer runs a workflow_dispatch job that runs the full build with secrets.
- Or require contributors to open PRs from branches inside the repo for full CI.

4. (Cautious) Use pull_request_target only for safe, non-code-executing steps
- pull_request_target has access to secrets but runs in the base repo context. Do NOT checkout and run untrusted PR code in that job.

### Suggested checklist for triage

 - [ ] Confirm the PR is from a fork (external contributor).
 - [ ] Confirm the secret NEXT_PUBLIC_CLERK_SIGN_IN_URL exists in Environments → staging.
 - [ ] Decide which mitigation to apply (skip, fail early, maintainer-trigger).
 - [ ] Update CI workflow with chosen change and add a test PR from a fork to confirm behavior.

### Additional notes

Even if environment secrets are configured correctly, they are intentionally not exposed to fork PRs for security reasons. The issue is not the secret location but the security model of GitHub Actions

--- a/.github/workflows/CI.yml
+++ b/.github/workflows/CI.yml
@@ -11,8 +11,6 @@ jobs:
     strategy:
       matrix:
         node-version: [20.x, 22.x]
-        # See supported Node.js release schedule at https://nodejs.org/en/about/releases/
-
     name: Build with ${{ matrix.node-version }}
     runs-on: ubuntu-latest
     environment: staging
@@ -27,6 +25,15 @@ jobs:
 
     steps:
       - uses: actions/checkout@v4
+
+      # If this is a PR from a fork, secrets (including environment secrets) are not available.
+      # Skip this secret-dependent build for forked PRs to avoid zod validation errors.
+      - name: Detect fork PR and skip secret-required build
+        if: ${{ github.event_name == 'pull_request' && github.event.pull_request.head.repo.full_name != github.repository }}
+        run: |
+          echo "This PR is from a fork. Environment secrets are not available; skipping secret-dependent build steps."
+          exit 0
+
       - name: Use Node.js ${{ matrix.node-version }}
         uses: actions/setup-node@v4
         with:
@@ -39,7 +46,6 @@ jobs:
     strategy:
       matrix:
         node-version: [20.x]
-
     name: Run all tests
     runs-on: ubuntu-latest
     environment: staging
@@ -56,6 +62,16 @@ jobs:
       - uses: actions/checkout@v4
         with:
           fetch-depth: 0 # Retrieve Git history, needed to verify commits
+
+      - name: Detect fork PR (for conditional steps)
+        id: detect_fork
+        run: |
+          if [ "${{ github.event_name }}" = "pull_request" ] && [ "${{ github.event.pull_request.head.repo.full_name }}" != "${{ github.repository }}" ]; then
+            echo "is_fork=true" >> $GITHUB_OUTPUT
+          else
+            echo "is_fork=false" >> $GITHUB_OUTPUT
+          fi
+
       - name: Use Node.js ${{ matrix.node-version }}
         uses: actions/setup-node@v4
         with:
@@ -63,7 +79,9 @@ jobs:
           cache: npm
       - run: npm ci
 
+      # Build Next.js for E2E tests and run E2E only when NOT a fork PR (these steps require secrets)
       - name: Build Next.js for E2E tests
+        if: steps.detect_fork.outputs.is_fork == 'false'
         run: npm run build
 
       - if: github.event_name == 'pull_request'
@@ -79,18 +97,12 @@ jobs:
       - name: Run unit tests
         run: npm run test -- --coverage
 
-      # - name: Upload coverage reports to Codecov
-      #   uses: codecov/codecov-action@v5
-      #   env:
-      #     CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
-
       - name: Install Playwright (used for Storybook and E2E tests)
         run: npx playwright install --with-deps
 
-      # - name: Run storybook tests
-      #   run: npm run test-storybook:ci
-
+      # Run E2E (and Percy) only when NOT a fork PR
       - name: Run E2E tests
+        if: steps.detect_fork.outputs.is_fork == 'false'
         run: npx percy exec -- npm run test:e2e
         env:
           PERCY_TOKEN: ${{ secrets.PERCY_TOKEN }}

